src/modules/mc_hover_thrust_estimator/MulticopterHoverThrustEstimator.cpp

```diff
@@ -113,56 +143,38 @@ void MulticopterHoverThrustEstimator::Run()
 
 	perf_begin(_cycle_perf);
 
-	if (_vehicle_land_detected_sub.updated()) {
-		vehicle_land_detected_s vehicle_land_detected;
-
-		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
-			_landed = vehicle_land_detected.landed;
-		}
-	}
-
 	if (_vehicle_status_sub.updated()) {
-		vehicle_status_s vehicle_status;
+		vehicle_status_s vehicle_status{};
 
 		if (_vehicle_status_sub.copy(&vehicle_status)) {
 			_armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
 		}
 	}
 
-	// always copy latest position estimate
-	vehicle_local_position_s local_pos{};
-
-	if (_vehicle_local_pos_sub.copy(&local_pos)) {
-		// This is only necessary because the landed
-		// flag of the land detector does not guarantee that
-		// the vehicle does not touch the ground anymore
-		// TODO: improve the landed flag
-		if (local_pos.dist_bottom_valid) {
-			_in_air = local_pos.dist_bottom > 1.f;
-		}
-	}
-
 	const float dt = (local_pos.timestamp - _timestamp_last) * 1e-6f;
 	_timestamp_last = local_pos.timestamp;
 
-	if (_armed && !_landed && (dt > 0.001f) && (dt < 1.f) && PX4_ISFINITE(local_pos.az)) {
+	if (_armed && _in_air && (dt > 0.001f) && (dt < 1.f) && PX4_ISFINITE(local_pos.az)) {
 
 		_hover_thrust_ekf.predict(dt);
 
-		vehicle_local_position_setpoint_s local_pos_sp;
+		vehicle_local_position_setpoint_s local_pos_sp{};
 
-		if (_vehicle_local_position_setpoint_sub.update(&local_pos_sp)) {
+		if (_vehicle_local_position_setpoint_sub.copy(&local_pos_sp)) {
```

Comment:
Do you want to fuse new accel Z with old thrust?
At the moment this is going to run on every `vehicle_local_position` publication (100 Hz), but `vehicle_local_position_setpoint` is only going to publish at 50 Hz in the correct mode.

Reply:
Nope, there is an early return if `vehicle_local_position_setpoint` isn't updated https://github.com/PX4/PX4-Autopilot/pull/16404/files#diff-e0c09a903a6632efbebee8a4054c24836d0fb34aa4bdad5d0c574b277954e3a1R130

Reply:
Ah right.